src/DshboardModule/Components/ADS/ADS.tsx

```diff
@@ -0,0 +1,580 @@
+        isActive: data.isActive// Get the original isActive value for the updated ad
+      };
+      const response = await axios.put(
+        `${baseUrl}/v0/admin/ads/${idAds}`,
+        payload, // Send the payload with discount and isActive
+        {
+          headers: {
+            Authorization: token,
+          },
+        }
+      );
+      console.log(response.data.message);
+      toast.success("Advertisement has been updated successfully.");
+      handleCloseDilaog();
+      fetchData();
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response) {
+        toast.error(error.response.data.message);
+      }
+    } finally {
+      setLoadingSubmit(false);
+    }
+    reset();
+  };
+  
+
+  useEffect(() => {
+    fetchData();
+    fetchDataAllRooms();
+  }, []);
+
+  const authContext = useContext(AuthContext);
+  if (!authContext) {
+    // Handle the case where AuthContext is null
+    return null;
+  }
+  const { baseUrl } = authContext;
+
+  return (
+    <>
+      {/* Modal For Delete  */}
+      <Modal
+        open={openModal}
+        onClose={handleCloseModal}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
+      >
+        <Box sx={style}>
+          <Box
+            sx={{
+              display: "flex",
+              justifyContent: "center",
+              alignItems: "center",
+              height: "100%",
+              marginY: "40px",
+            }}
+          >
+            <img src={ImageDelete} alt="" style={{ width: "30%" }} />
+          </Box>
+
+          <Typography
+            id="modal-modal-description"
+            sx={{ mt: 3, textAlign: "center" }}
+          >
+            Delete This Ads Room ?
+          </Typography>
+          <Typography
+            id="modal-modal-description"
+            sx={{ mt: 3, textAlign: "center" }}
+          >
+            are you sure you want to delete this item ? if you are sure just
+            click on delete it
+          </Typography>
+          <Box sx={{ textAlign: "end", m: 2 }}>
+            {loadingSubmit ? (
+              <CircularProgress size={24} />
+            ) : (
+              <Button
+                onClick={() => {
+                  DeleteData(idAds);
+                }}
+                variant="contained"
+              >
+                Delete
+              </Button>
+            )}
+          </Box>
+        </Box>
+      </Modal>
+
+      {/* end Modal Dialog */}
+      {/* Dialog for ADD ////////////////////Update  */}
+      <Dialog
+        open={openDialog}
+        onClose={handleCloseDilaog}
+        maxWidth="sm"
+        fullWidth
+      >
+        <DialogTitle sx={{ m: 2 }}>ADS</DialogTitle>
+        <DialogContent sx={{ mt: 1 }}>
+          <form onSubmit={handleSubmit(dialogMode === "add" ? onSubmit : updateAds)} style={{ width: "100%" }}>
+            <Grid container spacing={2}>
```

Comment:
We can render this grid as a form element and pass the attributes from the form element to it.